src/components/TokenForm/TokenForm.tsx

```diff
@@ -48,22 +49,43 @@ const TokenForm: FC<TokenFormProps> = ({
   const form = useFormContext<TokenFormValues>()
   const inputTokenAddress = form.watch("inputToken")
   const outputTokenAddress = form.watch("outputToken")
+  const amountIn = form.watch("amountIn")
 
   const tokenSelectField = useController({
     name: isWithdraw ? "outputToken" : "inputToken",
     control: form.control,
   })
 
+  const onClickMax = () => {
+    form.setValue("amountIn", inputTokenBalanceOrShare?.formatted ?? "0.0", {
+      shouldDirty: true,
+      shouldValidate: true,
+      shouldTouch: true,
+    })
+  }
+
   const { isConnected } = useAccount()
-  const { data: inputTokenBalance, isLoading: isLoadingInputTokenBalance } =
-    useTokenOrNativeBalance({ address: inputTokenAddress })
+  const {
+    data: inputTokenBalanceOrShare,
+    isLoading: isLoadingInputTokenBalanceOrShare,
+  } = useTokenOrNativeBalance({ address: inputTokenAddress })
   const { data: inputToken, isLoading: isLoadingInputToken } = useTokenOrNative(
     { address: inputTokenAddress }
   )
 
+  const hideMaxBtn =
```

Comment:
It might be easier to check to see if you should _show_ the button rather than hide it:

```ts
const showMaxBtn = inputTokenBalanceOrShare?.value?.gt(0) && inputTokenBalancerOrShare?.formatted !== amountIn
```

I might also disable it instead of hiding it completely, but I don't know what the designs are.

Reply:
Agreed, better disable it.